diff --git a/deletion_only_stale.cmake b/deletion_only_stale.cmake
index 4134fcf..975dc34 100644
--- a/deletion_only_stale.cmake
+++ b/deletion_only_stale.cmake
@@ -1,13 +1,7 @@
 if(NOT MSVC)
     add_compile_options(-Wall -Wextra -Wno-narrowing)
 else()
-    if (CMAKE_BUILD_TYPE MATCHES Debug)
-        add_compile_options(/MTd)
-    else()
-        add_compile_options(/MT)
-    endif()
-
     add_compile_options(/utf-8)
     add_compile_definitions(-D_CRT_SECURE_NO_WARNINGS=1)
 endif()
