--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -148,2 +148,4 @@
 endif()
+string(REPLACE ".dll" "" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
+list(APPEND LLVM_LIBRARIES "xml2.lib" "zstd.lib" "zlib.lib" "msvcrt.lib")

@@ -725,2 +727,4 @@
 else()
+target_compile_options(zigcpp PRIVATE -MD)
+add_compile_options(/MD /wd4715 /wd4291)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
@@ -847,1 +849,1 @@
-  target_link_libraries(zig1 LINK_PUBLIC m)
+  # target_link_libraries(zig1 LINK_PUBLIC m)
@@ -933,2 +936,3 @@
   -Dno-langref
+  --search-prefix "$ENV{PREFIX}/Library/lib"
   --maxrss 7000000000
@@ -949 +1010 @@
-if(ZIG_STATIC AND NOT MSVC)
+if(ZIG_STATIC AND NOT MSVC)
